### DISCUSSION TITLE
IsNullOrEmptyConverter uses string.IsNullOrWhiteSpace?

### DISCUSSION THREAD

**Discussion by gdang** [xamarin/XamarinCommunityToolkit#1738] [Q&A]
I'm using the `IsNullOrEmptyConverter` in my project to show a UI element when an input string is null or empty, but when I enter whitespace characters only, the UI element is still visible.

So I looked at the source code, and to my surprise, the converter is implemented using `string.IsNullOrWhiteSpace`:
https://github.com/xamarin/XamarinCommunityToolkit/blob/main/src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/IsNullOrEmptyConverter.shared.cs#L24

I'm not sure if that is the correct intention of the code. If it is, then the converter should be renamed to **IsNullOrWhitespaceConverter**.

**Comment by bijington** [comment]
Thank you for pointing this out. It certainly looks like a bug to me! Would you be willing to raise it as a bug?

I would suggest we end up with 2 converters 1 to cover the existing white space check and one that does not.

**Reply by pictos** [reply]
We can add a bool property that `IgnoreWitheSpaces` and handle that in the same convert. What do you think about it?

**Reply by bijington** [reply]
Yes that could work

**Reply by gdang** [reply]
I prefer having two separate converters, but the `IgnoreWhiteSpaces` property is fine.

I'll open an issue.

**Reply by bijington** [reply]
Two separate ones may be clearer in terms of use, we could even have a third `IsNullConverter`
